extension/js/common/api/email-provider/gmail/gmail.ts

```diff
@@ -33,6 +33,10 @@ export class Gmail extends EmailProviderApi implements EmailProviderInterface {
     }
   };
 
+  public threadIdGet = async (messageId: string): Promise<GmailRes.GmailThreadIdQuery> => {
+    return await Google.gmailCall<GmailRes.GmailThreadIdQuery>(this.acctEmail, 'messages', { method: 'GET', data: { q: `rfc822msgid:${messageId}` } });
+  };
```

Comment:
we already have `msgList` which performs the same api call, let's just use it with `q` param set to `rfc822msgid:${messageId}`

Reply:
Thanks - I'll remove threadIdGet now and the related typings i made and I'll use the one that already exists.